src/azul/plugins/__init__.py

```diff
@@ -723,16 +745,22 @@ def validate_version(self, version: str) -> None:
         raise NotImplementedError
 
 
-@attr.s(auto_attribs=True, kw_only=True)
-class RepositoryFileDownload(metaclass=ABCMeta):
-    #: The UUID of the file to be downloaded
-    file_uuid: str
+@attr.s(auto_attribs=True, frozen=True, kw_only=True)
+class File(SerializableAttrs):
+    """
+    A non-abstract base class used in situations where there isn't enough
+    context to construct an instance from a subclass of :class:`PluginFile`.
```

Comment:
Too hand-wavy for my taste.

```suggestion
    A reference to a data file in the repository.
```